web-common/src/features/dashboards/time-series/MetricsTimeSeriesCharts.svelte

```diff
@@ -204,114 +204,117 @@
   };
 </script>
 
-<TimeSeriesChartContainer
-  enableFullWidth={Boolean(expandedMeasureName)}
-  end={endValue}
-  start={startValue}
-  {workspaceWidth}
->
-  <div class="bg-white sticky top-0 flex pl-1 z-10">
-    {#if expandedMeasureName}
-      <BackToOverview {metricViewName} />
-    {:else}
-      <SeachableFilterButton
-        label="Measures"
-        on:deselect-all={setAllMeasuresNotVisible}
-        on:item-clicked={toggleMeasureVisibility}
-        on:select-all={setAllMeasuresVisible}
-        selectableItems={$showHideMeasures.selectableItems}
-        selectedItems={$showHideMeasures.selectedItems}
-        tooltipText="Choose measures to display"
-      />
-    {/if}
-  </div>
-  <div class="bg-white sticky left-0 top-0 overflow-visible z-10">
-    <!-- top axis element -->
-    <div />
-    <MeasureZoom {metricViewName} />
-    {#if $dashboardStore?.selectedTimeRange && startValue && endValue}
-      <SimpleDataGraphic
-        height={26}
-        overflowHidden={false}
-        top={29}
-        bottom={0}
-        xMin={startValue}
-        xMax={endValue}
-      >
-        <Axis superlabel side="top" placement="start" />
-      </SimpleDataGraphic>
-    {/if}
-  </div>
-  <!-- bignumbers and line charts -->
-  {#if renderedMeasures.length}
-    <!-- FIXME: this is pending the remaining state work for show/hide measures and dimensions -->
-    {#each renderedMeasures as measure (measure.name)}
-      <!-- FIXME: I can't select the big number by the measure id. -->
-      <!-- for bigNum, catch nulls and convert to undefined.  -->
-      {@const bigNum = totals?.[measure.name] ?? undefined}
-      {@const comparisonValue = totalsComparisons?.[measure.name]}
-      {@const isValidPercTotal = $isMeasureValidPercentOfTotal(measure.name)}
-      {@const comparisonPercChange =
-        comparisonValue && bigNum !== undefined && bigNum !== null
-          ? (bigNum - comparisonValue) / comparisonValue
-          : undefined}
-      <MeasureBigNumber
-        {measure}
-        value={bigNum}
-        isMeasureExpanded={!!expandedMeasureName}
-        {showComparison}
-        comparisonOption={$timeControlsStore?.selectedComparisonTimeRange?.name}
-        {comparisonValue}
-        {comparisonPercChange}
-        status={$timeSeriesDataStore?.isFetching
-          ? EntityStatus.Running
-          : EntityStatus.Idle}
-        on:expand-measure={() => {
-          metricsExplorerStore.setExpandedMeasureName(
-            metricViewName,
-            measure.name,
-          );
-        }}
-      />
+{#if $timeControlsStore.ready}
```

Comment:
Did you investigate whether this conditional can be pushed deeper into the component tree?

I'm fine with going ahead and merging this to clear this bug as a quick stop gap fix, but I suspect we can be more granular here. I'll note that the whole `time-series` folder is one of the ones that [still needs to be cleaned up for strict null checks](https://github.com/rilldata/rill/issues/3424), which is presumably how this bug is possible in the first place.

So I'll happy be happy to dig deeper and clean this up when I get to this file, but in the meantime, I propose adding a FIXME comment before this condition that links back to this PR and the issue and explains briefly that we added this non-granular quick fix, but that we think we can do better. A little reminder like that will help me to be less afraid to refactor this guard out when I get to cleaning up this file

Reply:
Good call. I just made a more precise conditional, which removed the culpable type error 👍

Reply:
love it! 👍